turbinia/evidence.py

```diff
@@ -282,12 +304,31 @@ def __init__(self, project=None, zone=None, disk_name=None, *args, **kwargs):
     super(GoogleCloudDisk, self).__init__(*args, **kwargs)
     self.cloud_only = True
 
+    self._attached_path = None
+
   def _preprocess(self):
-    self.local_path = google_cloud.PreprocessAttachDisk(self.disk_name)
+    self._attached_path = google_cloud.PreprocessAttachDisk(self.disk_name)
+    self.loopdevice_path = mount_local.PreprocessLosetup(self._attached_path)
+
+    # If the mounting step fails, at least local_path still points to a device
+    # and can be picked up by PlasoTask for example
+    self.local_path = self.loopdevice_path
+    try:
+      self.mount_path = mount_local.PreprocessMountDisk(
+          self.loopdevice_path, self.mount_partition)
+      self.local_path = self.mount_path
+      self.is_mounted = True
+    except TurbiniaException as e:
+      log.error(
+          'Could not mount partition {0:d} for GoogleCloudDisk {1:s}: '
+          '{2!s}'.format(self.mount_partition, self.disk_name, e))
 
   def _postprocess(self):
-    google_cloud.PostprocessDetachDisk(self.disk_name, self.local_path)
-    self.local_path = None
+    if self.is_mounted:
+      mount_local.PostprocessUnmountPath(self.mount_path)
+      self.is_mounted = False
+    mount_local.PostprocessDeleteLosetup(self.loopdevice_path)
+    google_cloud.PostprocessDetachDisk(self.disk_name, self._attached_path)
```

Comment:
Should we still set `self.local_path = None` here?

Reply:
So turns out setting self.local_path to None is a bad idea, because IIUC one Evidence object can be postprocessed, but later on re-preprocessed. If we unset local_path, we can't re-preprocess